async/kafka/group/group.go

```diff
@@ -0,0 +1,179 @@
+package group
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"runtime"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/async"
+	"github.com/beatlabs/patron/async/kafka"
+	"github.com/beatlabs/patron/errors"
+	"github.com/beatlabs/patron/log"
+	"github.com/opentracing/opentracing-go"
+)
+
+// Factory definition of a consumer factory.
+type Factory struct {
+	name    string
+	group   string
+	topic   string
+	brokers []string
+	oo      []kafka.OptionFunc
+}
+
+// New constructor.
+func New(name, group, topic string, brokers []string, oo ...kafka.OptionFunc) (*Factory, error) {
+
+	if name == "" {
+		return nil, errors.New("name is required")
+	}
+
+	if group == "" {
+		return nil, errors.New("group is required")
+	}
+
+	if len(brokers) == 0 {
+		return nil, errors.New("provide at least one broker")
+	}
+
+	if topic == "" {
+		return nil, errors.New("topic is required")
+	}
+
+	return &Factory{name: name, group: group, topic: topic, brokers: brokers, oo: oo}, nil
+}
+
+// Create a new consumer.
+func (f *Factory) Create() (async.Consumer, error) {
```

Comment:
We should strive to follow the rule:
`Accept interfaces return structs`
https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8

Reply:
If we do not return the `consumer` hidden by the interface we have to make the `consumer` struct public is that something we want :question:

Reply:
Correct, I did not take into account the consumer factory.